### ISSUE TITLE
connect() failed (111: Connection refused) while connecting to upstream

### ISSUE THREAD

**Issue by lippoliv** [bitwarden/server#2544]
### Steps To Reproduce

1. Setup bitwarden docker with version 2022.9.1
2. Put it behind an nginx reverse proxy
3. Update with shell script to newest version (2022.12.0)

### Expected Result

You can access bitwarden UI.

### Actual Result

You get a `502 Bad Gateway`

### Additional Context

I have a **Families Organization (Annually)** Subscription

**nginx site conf**
```
server {
        listen 80;
        listen [::]:80;

        access_log /var/log/nginx/bitwarden.access.log;
        error_log /var/log/nginx/bitwarden.error.log;

        server_name bitwarden....;

        include global/letsencrypt-auth.conf;

        location / {
                return 301 https://$host$request_uri;
        }
}
server {
        resolver 127.0.0.11 valid=30s;

        listen 443 ssl;
        listen [::]:443 ssl;

        access_log /var/log/nginx/bitwarden.access.log;
        error_log /var/log/nginx/bitwarden.error.log;

        client_max_body_size 1G;

        ssl on;
        ssl_certificate /etc/letsencrypt/live/bitwarden/fullchain.pem;
        ssl_certificate_key /etc/letsencrypt/live/bitwarden/privkey.pem;

        ssl_session_cache  builtin:1000  shared:SSL:10m;
        ssl_protocols  TLSv1 TLSv1.1 TLSv1.2;
        ssl_ciphers HIGH:!aNULL:!eNULL:!EXPORT:!CAMELLIA:!DES:!MD5:!PSK:!RC4;
        ssl_prefer_server_ciphers on;

        server_name bitwarden...;

        include global/letsencrypt-auth.conf;

        location / {
                proxy_set_header    X-Real-IP          $remote_addr;
                proxy_set_header    X-Forwarded-For    $remote_addr;
                proxy_set_header    Host               $host;
                proxy_set_header    X-Forwarded-Proto  $scheme;

                set $upstream http://bitwarden:8080;
                proxy_pass $upstream;
        }
}
```

**bitwarden.error.log**
```
2023/01/06 12:39:05 [error] 52#52: *8 connect() failed (111: Connection refused) while connecting to upstream, client: ..., server: bitwarden, request: "GET /favicon.ico HTTP/1.0", upstream: "http://172.18.0.21:5000/favicon.ico", host: "bitwarden...", referrer: "https://bitwarden..../"
```

### Build Version

2022.12.0

### Environment

Self-Hosted

### Environment Details

- Operating system: Linux Debian
- Environment: Docker
- Hardware: Intel(R) Core(TM) i7-8700 CPU @ 3.20GHz, 128 GB RAM

### Issue Tracking Info

- [X] I understand that work is tracked outside of Github. A PR will be linked to this issue should one be opened to address it, but Bitwarden doesn't use fields like "assigned", "milestone", or "project" to track progress.

**Comment by lippoliv** [comment]
Additional info: This setup worked for over a year, across multiple updates.

**Comment by lippoliv** [comment]
I did some analysis. Found out, that while many other docker-applications on that host, behind the same reverse proxy, where running without issues, at least Sentry wasn't running as well.

Sentry and Bitwarden have one thing in common: My nginx container is connected to another nginx container of the docker compose projects which then redirect the traffic to particular containers.

Here's how I fixed it, I post it even if it is very specific:

I connect my nginx reverse proxy to the applications via a docker network. I map it to Bitwarden by modifying the docker-compose yml (not ideal, but "override" doesn't work here afaik).

- Reinstall docker-compose (ensure to have newest version, in my case I still had `docker-compose` installed, removed it)
- stop all docker containers
- flush all iptable configuration (do a backup first)
- restart docker service
- In the nginx reverse proxy, set version to (at least) 3.5
- ensure to initialize the network this way
    ```
    networks:
      net:
        name: nginx_net
        driver: bridge
    ```
- use this external network this way
    ```
    networks:
      default:
        internal: true
      public:
        internal: false
      # Added 2 lines
      nginx_net:
        external: true
    ```

I know it's very special setup, probably. Maybe someone else using the same configuration can fix his issues too by reading this.

**Comment by lippoliv** [comment]
The problem occures again after updating to 2023.1.0. Is it really a problem in my setup or maybe others have similar issues?

**Comment by rheaalleen** [comment]
Nope, also happened to me a few minutes ago, from working instance to 502 in web interface and connection refused in the container logs.

I switch branch to dev and not beta, restarted and its working at the moment

**Comment by DanielGilbert** [comment]
I have the same issue on my install - wanted to check the iterations in my account, and was greeted with a 502. I'm running the bitwarden nginx behind Traefik, so also a proxied nginx.

**Comment by closebot-bw** [comment]
⚠️ **Stale Issue Notice**

This issue has been automatically marked as stale due to inactivity. It will be closed in **2 weeks** (October 15, 2025) if no further activity occurs.

If this issue is still relevant and you would like to keep it open, please:
- Comment on this issue to show continued interest
- Provide any additional information or updates
- Confirm that the issue still exists in the latest version

Thank you for your contribution to this project! 🙏

**Comment by closebot-bw** [comment]
🔒 **Issue Closed Due to Inactivity**

This issue has been automatically closed due to lack of activity for an extended period. We periodically review and close inactive issues to help maintain our issue tracker and focus on current priorities.

**If this issue is still relevant:**
- Please create a new issue with updated information
- Include steps to reproduce the problem if it's a bug report
- Mention if this issue still occurs in the latest version

Thank you for your contribution to this project. Your feedback helps us improve! 🙏
